### PR TITLE
Move stuff from unicore.distribute over here.

### PR DESCRIPTION
**Pull request by smn**

### CODE REVIEW

**Comment by smn** [comment]
a `unicore.webhooks` appears!
@miltontony, @Rizziepit & @Zolisa001 please review?

File: unicore/webhooks/views.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                          event_type=event_type,
+                          active=active)
+        data = webhook.to_dict()
+        DBSession.add(webhook)
+        self.request.registry.notify(WebhookCreated(webhook, self.request))
+        return data
+
+    @view(renderer='json', schema=WebhookSchema)
+    def put(self):
+        webhook = self.webhooks. \
```
Comment (miltontony) [inline unicore/webhooks/views.py:78]: `null` check is required here

``` python
if webhook is None:
    raise NotFound
```

File: unicore/webhooks/views.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            filter(Webhook.uuid == self.request.matchdict['uuid']). \
```
Comment (miltontony) [inline unicore/webhooks/views.py:79]: _optional_: there's a neater shorthand `filter_by(uuid=self.request.matchdict['uuid'])`

**Comment by Zolisa001** [comment]
Reviewing done.

File: unicore/webhooks/views.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+
+# Raise SQLAlchemy exceptions as HTTP 404s
+@view_config(context=NoResultFound)
+@view_config(context=MultipleResultsFound)
+def sqlachemy_error(exception, request):
+    raise httpexceptions.HTTPNotFound()
+
+
+@resource(collection_path='/hooks.json', path='/hooks/{uuid}.json')
```
Comment (miltontony) [inline unicore/webhooks/views.py]: I think as a team we need to standardize on the format of our rest URLs. I think we should either use `/hooks.json` or `/hooks`, but we can't alternate between the 2.

Comment (smn) [inline unicore/webhooks/views.py]: Ok, tell me what your preference is and then I'll update accordingly.

File: unicore/webhooks/views.py (hunk 1 above)
Comment (smn) [inline unicore/webhooks/views.py:78]: I was thinking [this](https://github.com/universalcore/unicore.webhooks/pull/1/files#diff-7e36f05d66e73ec236735b69f6e17730R24) did that?

Comment (miltontony) [inline unicore/webhooks/views.py:78]: Cool, I think it does. @Rizziepit can you confirm?

File: unicore/webhooks/views.py (hunk 3 above)
Comment (miltontony) [inline unicore/webhooks/views.py]: Lets go with just `/hooks` and `/hooks/{uuid}`

File: unicore/webhooks/views.py (hunk 2 above)
Comment (smn) [inline unicore/webhooks/views.py:79]: Thanks, still finding my sqlalchemy feet.

File: unicore/webhooks/views.py (hunk 3 above)
Comment (smn) [inline unicore/webhooks/views.py]: ok I'll update in `unicore.distribute` too and do a new release.

**Comment by miltontony** [comment]
:+1:

File: unicore/webhooks/views.py (hunk 1 above)
Comment (Rizziepit) [inline unicore/webhooks/views.py:78]: Yes, it does (fwiw post-merge)

**Pull request merged**